### PR TITLE
Add measurement functions for densitytensor

### PR DESCRIPTION
**Pull request by SamDuffield**
- Cleaned up the tests (removed some imports etc)
- Added `densitytensor_to_measurement_probabilities` and `densitytensor_to_measured_densitytensor`, these are almost certainly coded suboptimally but I just want to get the API sorted for now

### CODE REVIEW

File: qujax/density_matrix.py [hunk 1]
```diff
@@ -215,3 +215,50 @@ def no_params_to_densitytensor_func(densitytensor_in: jnp.ndarray = None) -> jnp
...
+    n_qubits = densitytensor.ndim // 2
+    n_qubits_measured = len(qubit_inds)
+    qubit_inds_trace_out = [i for i in range(n_qubits) if i not in qubit_inds]
+    return jnp.diag(partial_trace(densitytensor, qubit_inds_trace_out).reshape(2 * n_qubits_measured,
+                                                                               2 * n_qubits_measured)).real
+
+
+def densitytensor_to_measured_densitytensor(densitytensor: jnp.ndarray,
+                                            qubit_inds: Sequence[int],
+                                            measured_int: int) -> jnp.ndarray:
```
Comment (gamatos) [inline qujax/density_matrix.py:244]: Would it make more sense for `measured_int` to be a bit string instead?

Comment (SamDuffield) [reply qujax/density_matrix.py:244]: Maybe we could accept either?

And if a bitstring is given, we convert it to an int

Comment (gamatos) [reply qujax/density_matrix.py:244]: Sounds reasonable, and that only uses information about the shape, so the function should still be JIT'able

**Review by gamatos — APPROVED** [review]
Looks good. Added a test case to test the bit string argument to `densitytensor_to_measured_densitytensor`, if that is good it can be merged

**Pull request merged**